src/evict/evict_page.c

```diff
@@ -286,8 +286,15 @@ __evict_delete_ref(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
          * This will consume the deleted ref (and eventually free it). If the reverse split can't
          * get the access it needs because something is busy, be sure that the page still ends up
          * marked deleted.
+         *
+         * Don't do it if we are a VLCS tree and the child we're deleting is the leftmost child. The
+         * reverse split will automatically remove the page entirely, creating a namespace gap at
+         * the beginning of the internal page, and that leaves search nowhere to go. Note that the
+         * situation will be handled safely if another child gets deleted, or if eviction comes for
+         * a visit.
```

Comment:
Good point. Added in [c265cda](https://github.com/wiredtiger/wiredtiger/pull/8129/commits/c265cda33a2ef72d4a6f23d411ced096da3f9877).

(Along with another one for reverse splits overall, since we apparently aren't counting them and the one stat probably isn't that much use without the other.)